examples/nanocoap_server/main.c

```diff
@@ -41,8 +41,22 @@ int main(void)
     xtimer_sleep(3);
 
     /* print network addresses */
-    puts("Configured network interfaces:");
-    _gnrc_netif_config(0, NULL);
+    gnrc_netif_t *netif = NULL;
+    while ((netif = gnrc_netif_iter(netif))) {
+        ipv6_addr_t ipv6_addrs[CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF];
+        int res = gnrc_netapi_get(netif->pid, NETOPT_IPV6_ADDR, 0, ipv6_addrs,
+                                  sizeof(ipv6_addrs));
+
+        if (res < 0) {
+            continue;
+        }
+        for (unsigned i = 0; i < (unsigned)(res / sizeof(ipv6_addr_t)); i++) {
+            char ipv6_addr[IPV6_ADDR_MAX_STR_LEN];
+
+            ipv6_addr_to_str(ipv6_addr, &ipv6_addrs[i], IPV6_ADDR_MAX_STR_LEN);
+            printf("My address is %s\n", ipv6_addr);
+        }
+    }
```

Comment:
I think it should also be possible to use [`netif_iter()`](https://api.riot-os.org/group__net__netif.html#ga1688d9ea3c9c2d8d3bd2c8e29b0dc06c) and [`netif_get_opt()`](https://api.riot-os.org/group__net__netif.html#gae51c651b0826888c7b18507a762100d0) instead of the GNRC variants to do this stack independent.